packages/gestalt/src/Tabs.js

```diff
@@ -73,45 +128,137 @@ function Tab({
           rounding="pill"
           userSelect="none"
         >
-          <Text color={isActive ? 'white' : 'darkGray'} weight="bold" overflow="noWrap">
-            {children}
-          </Text>
-          {indicator === 'dot' && <Circle />}
+          <Flex alignItems="center" gap={2} justifyContent="center">
+            <Text color={isActive ? 'white' : 'darkGray'} weight="bold" overflow="noWrap">
+              {text}
+            </Text>
+
+            {indicator === 'dot' && <Dot />}
+          </Flex>
         </Box>
       </Link>
     </Box>
   );
 }
 
+const TAB_ROUNDING = 2;
+const TAB_INNER_PADDING = 1;
+
+function TabV2({
+  href,
+  indicator,
+  id,
+  index,
+  isActive,
+  onChange,
+  // No longer supported, will be removed when this variant ships
+  // eslint-disable-next-line no-unused-vars
+  size,
+  text,
+}: {|
+  ...TabType,
+  index: number,
+  isActive: boolean,
+  onChange: OnChangeHandler,
+  size: 'md' | 'lg',
+|}) {
+  const [hovered, setHovered] = useState(false);
+  const [focused, setFocused] = useState(false);
+  const [pressed, setPressed] = useState(false);
+
+  let color = 'white';
+  if (pressed) {
+    color = 'lightWash';
+  } else if ((hovered || focused) && !isActive) {
+    color = 'lightGray';
+  }
+
+  return (
+    <Box id={id} paddingY={3}>
```

Comment:
@dangerismycat In gestalt we tried to avoid extra padding around components. It make the components less composable, e.g. when `<Tabs />` is part of a `<Flex />` with a `gap`?

Totally fine in the docs example to wrap the `<Tabs />` in a `<Box paddingY={3} />` but I would avoid making the outside padding part of the component.

![image](https://user-images.githubusercontent.com/127199/112697943-d5747980-8e45-11eb-8fb7-074edef0e9b1.png)

Reply:
Makes sense. That vertical padding was part of the design — though the way this is implemented, there might be too much padding if the tabs wrap. Asking @anniemochi on Slack now

Reply:
Yep, this implementation is approved by Annie (as opposed to setting the vertical padding on the Tabs component as a whole, so that the rows would be closer vertically)